### PR TITLE
[RuntimePermissions] Modernize phone app

### PR DESCRIPTION
**Pull request by alexvanyo**
Updates the phone portion of the `RuntimePermissions` app, converting from Java to Kotlin, introducing coroutines, and other best practices.

I also switched the phone permission from storage to the cellular information, since the old behavior (showing all of the folders in the emulated storage) is deprecated.

### CODE REVIEW

File: RuntimePermissionsWear/Application/build.gradle [hunk 1]
```diff
@@ -20,17 +20,23 @@ plugins {
 }
 
 android {
-    compileSdkVersion 30
+    compileSdkVersion 31
```
Comment (codingjeremy) [inline RuntimePermissionsWear/Application/build.gradle:23]: Is this just so it will build right now?

**Review by codingjeremy — APPROVED** [review]
Just some minor nits.

File: RuntimePermissionsWear/Application/src/main/java/com/example/android/wearable/runtimepermissions/IncomingRequestPhoneService.kt [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+            }
+        }
+    }
+
+    private suspend fun promptUserForPhonePermission(nodeId: String) {
+        val phoneInfoPermissionApproved =
+            (
+                ActivityCompat.checkSelfPermission(this, phoneSummaryPermission)
+                    == PackageManager.PERMISSION_GRANTED
+                )
```
[the remark this answers is not in the document]
Comment (alexvanyo) [reply RuntimePermissionsWear/Application/src/main/java/com/example/android/wearable/runtimepermissions/IncomingRequestPhoneService.kt:75]: Oh yeah, this looks weird... it looks like there's an unnecessary set of parentheses around the condition. I'll remove that.

File: RuntimePermissionsWear/Application/build.gradle (hunk 1 above)
Comment (alexvanyo) [reply RuntimePermissionsWear/Application/build.gradle:23]: Ah, here I'm updating the phone app. So no reason to avoid going to API 31!

**Comment by alexvanyo** [comment]
Thanks @codingjeremy !

**Pull request merged**